src/taskgraph/util/vcs.py

```diff
@@ -210,14 +216,19 @@ def _files_template(self, diff_filter):
             template += "{file_mods % '{file}\\n'}"
         return template
 
-    def get_changed_files(self, diff_filter="ADM", mode="unstaged", rev=None):
+    def get_changed_files(
+        self, diff_filter="ADM", mode="unstaged", rev=None, base_rev=None
+    ):
         if rev is None:
+            if base_rev is not None:
+                raise ValueError("Cannot specify `base_rev` without `rev`")
             # Use --no-status to print just the filename.
             df = self._format_diff_filter(diff_filter, for_status=True)
             return self.run("status", "--no-status", f"-{df}").splitlines()
         else:
             template = self._files_template(diff_filter)
-            return self.run("log", "-r", rev, "-T", template).splitlines()
+            revision_argument = rev if base_rev is None else f"{base_rev}~-1::{rev}"
```

Comment:
Huh, TIL you can do `~-1`. I think just `~1` is equivalanet fwiw.

Reply:
Same here, TIL when I stumbled upon that problem 🙂 They're not totally equivalent. Here's the relation between between all 3:

```
tip
...
base_rev~-1
base_rev
base_rev~1
...
older_revisions
```

I confess I'm not 100% sure `~-1` is the right way to go. It works in the unit tests but I wonder how it behaves when a `base_rev` has more than a single child revision - which is quite common on `try`. 

The good news is: this code won't impact our `hg` repos yet. `hg` still uses `json-automationrelevance` while `git` gets tested in the real world. If that looks good enough, I'll perform another round of tests on hg before stopping to rely on `json-automationrelevance`